### PR TITLE
[SPARK-47610][CORE] Always set `io.netty.tryReflectionSetAccessible=true`

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Always set `io.netty.tryReflectionSetAccessible=true`

### Why are the changes needed?

Arrow requires `-Dio.netty.tryReflectionSetAccessible=true` for JDK9+, see details in ARROW-7223. SPARK-29924 (fixed in 3.0.0) added document to guide users to add such JavaOpts manually, as Arrow is a Spark built-in component, and later we added such Java options to the building system(Maven, SBT, and PySpark test suite) manually. Now Spark requires JDK 17+, I think we can add such Java options by default to reduce disturbing users.

### Does this PR introduce _any_ user-facing change?

Yes, no impacts for those users who manually added `io.netty.tryReflectionSetAccessible=true`, but makes life easier for new Spark users. 

### How was this patch tested?

Pass GA.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: launcher/src/main/java/org/apache/spark/launcher/JavaModuleOptions.java [hunk 1]
```diff
@@ -42,7 +42,8 @@ public class JavaModuleOptions {
       "--add-opens=java.base/sun.security.action=ALL-UNNAMED",
       "--add-opens=java.base/sun.util.calendar=ALL-UNNAMED",
       "--add-opens=java.security.jgss/sun.security.krb5=ALL-UNNAMED",
-      "-Djdk.reflect.useDirectMethodHandle=false"};
+      "-Djdk.reflect.useDirectMethodHandle=false",
+      "-Dio.netty.tryReflectionSetAccessible=true"};
```
Comment (pan3793) [inline launcher/src/main/java/org/apache/spark/launcher/JavaModuleOptions.java:46]: The reviewers may argue that the newly added arg does not match the class name`JavaModuleOptions`, but considering that SPARK-40729 already got in, we can do refactor later

**Comment by pan3793** [comment]
cc @dongjoon-hyun @HyukjinKwon @LuciferYang

File: sql/hive/pom.xml [hunk 2]
```diff
@@ -233,7 +233,7 @@
         <artifactId>scalatest-maven-plugin</artifactId>
         <configuration>
           <!-- Specially disable assertions since some Hive tests fail them -->
-          <argLine>-da -Xmx4g -XX:ReservedCodeCacheSize=${CodeCacheSize} ${extraJavaTestArgs} -Dio.netty.tryReflectionSetAccessible=true</argLine>
```
Comment (LuciferYang) [inline sql/hive/pom.xml:236]: Perhaps it cannot be deleted here, just like `extraJavaTestArgs`, some test cases do not use `JavaModuleOptions` to supplement default parameters, same as `SparkBuild.scala`

Comment (pan3793) [reply sql/hive/pom.xml:236]: I just moved it into `extraJavaTestArgs`

Comment (LuciferYang) [reply sql/hive/pom.xml:236]: OK

**Review by LuciferYang — APPROVED** [review]
+1, LGTM (pending tests)

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Thank you, @pan3793 and @LuciferYang .

**Comment by LuciferYang** [comment]
Merged into master for Spark 4.0. Thanks @pan3793 and @dongjoon-hyun ~

**Pull request closed without merging**